src/common/meta/src/key/flownode_task.rs

```diff
@@ -0,0 +1,164 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use futures::stream::BoxStream;
+use futures::StreamExt;
+use snafu::OptionExt;
+
+use crate::error::{self, Result};
+use crate::key::{
+    PartitionId, TableMetaKey, TaskId, FLOWNODE_TASK_KEY_PATTERN, FLOWNODE_TASK_KEY_PREFIX,
+};
+use crate::kv_backend::txn::{Txn, TxnOp};
+use crate::kv_backend::KvBackendRef;
+use crate::range_stream::{PaginationStream, DEFAULT_PAGE_SIZE};
+use crate::rpc::store::RangeRequest;
+use crate::rpc::KeyValue;
+use crate::FlownodeId;
+
+/// The key of mapping [FlownodeId] to [TaskId].
+pub struct FlownodeTaskKey {
+    flownode_id: FlownodeId,
+    task_id: TaskId,
+    partition_id: PartitionId,
+}
+
+impl FlownodeTaskKey {
+    /// Returns a [FlownodeTaskKey] with the specified `flow_node` and `task_id`.
+    pub fn new(flownode_id: FlownodeId, task_id: TaskId, partition_id: PartitionId) -> Self {
+        Self {
+            flownode_id,
+            task_id,
+            partition_id,
+        }
+    }
+
+    fn prefix(flownode_id: FlownodeId) -> String {
+        format!("{}/{flownode_id}", FLOWNODE_TASK_KEY_PREFIX)
+    }
+
+    /// The prefix used to retrieve all [FlownodeTaskKey]s with the specified `flownode_id`.
+    pub fn range_start_key(flownode_id: FlownodeId) -> String {
+        format!("{}/", Self::prefix(flownode_id))
+    }
+
+    /// Strips the [TaskId] from bytes.
+    pub fn strip_task_id(raw_key: &[u8]) -> Result<TaskId> {
+        let key = String::from_utf8(raw_key.to_vec()).map_err(|e| {
+            error::InvalidTableMetadataSnafu {
+                err_msg: format!(
+                    "FlownodeTaskKey '{}' is not a valid UTF8 string: {e}",
+                    String::from_utf8_lossy(raw_key)
+                ),
+            }
+            .build()
+        })?;
+        let captures =
+            FLOWNODE_TASK_KEY_PATTERN
+                .captures(&key)
+                .context(error::InvalidTableMetadataSnafu {
+                    err_msg: format!("Invalid FlownodeTaskKey '{key}'"),
+                })?;
+        // Safety: pass the regex check above
+        let task_id = captures[2].parse::<TaskId>().unwrap();
+        Ok(task_id)
+    }
+}
+
+impl TableMetaKey for FlownodeTaskKey {
+    fn as_raw_key(&self) -> Vec<u8> {
+        format!(
+            "{FLOWNODE_TASK_KEY_PREFIX}/{}/{}/{}",
+            self.flownode_id, self.task_id, self.partition_id,
+        )
+        .into_bytes()
+    }
+}
+
+/// The manager of [FlownodeTaskKey].
+pub struct FlownodeTaskManager {
+    kv_backend: KvBackendRef,
+}
+
+/// Decodes `KeyValue` to ((),[TaskId])
+pub fn flownode_task_key_decoder(kv: KeyValue) -> Result<((), TaskId)> {
```

Comment:
Any particular thought about returning a "()"? It looks useless to me.

Reply:
> Any particular thought about returning a "()"? It looks useless to me.

This will be refactored in another PR.